src/component/chatting/talk.tsx

```diff
@@ -0,0 +1,183 @@
+        <Chat>
+          {
+            <Time>
+              {chatTime(chattingList[chattingRoomId].message[messageId].date)}
+            </Time>
+          }
+          <NowUser>
+            {chattingList[chattingRoomId].message[messageId].addText}{' '}
+          </NowUser>
+        </Chat>
+      ) : (
+        <Chat>
+          <PartnerImage
+            src={`/images/${chattingList[chattingRoomId].message[messageId].userNum}.jpg`}
+          />
+          <Contents
+            onMouseEnter={() => {
+              setShowBtn(true);
+            }}
+            onMouseLeave={() => {
+              setShowBtn(false);
+            }}
+          >
+            <PartnerName>
+              {
+                userList[
+                  chattingList[chattingRoomId].message[messageId].userNum
+                ].userName
+              }
+            </PartnerName>
+            <Content>
+              <CounterPart>
+                {chattingList[chattingRoomId].message[messageId].addText}
+              </CounterPart>
+              {
+                <Time>
+                  {chatTime(
+                    chattingList[chattingRoomId].message[messageId].date
+                  )}
+                </Time>
+              }
+              <div className={`list ${showBtn ? 'show' : 'hide'}`}>
+                <Button onClick={() => setSympathy(true)}>
+                  <div className="heart">{'♥'}</div>
+                </Button>
+              </div>
+            </Content>
+            <div className={`heart ${sympathy ? 'show' : 'hide'}`}>
+              <Heart
+                src={`/images/sympathy.png`}
+                onClick={() => setSympathy(false)}
+              ></Heart>
+            </div>
+          </Contents>
+        </Chat>
+      )}
+    </>
+  );
+};
+const Heart = styled.img`
+  height: 0.8rem;
+  background-color: white;
+  margin: 0.3rem 0 0 0.5rem;
+  padding: 0.3rem;
+  border-radius: 0.3rem;
+  :hover {
+    filter: opacity(50%);
+    cursor: pointer;
+  }
+`;
+const Button = styled.button`
+  background-color: rgb(234 234 234 / 76%);
+  border: none;
+  padding: 0;
+  width: 1.5rem;
+  height: 1.5rem;
+  border-radius: 0.3rem;
+  .heart {
+    width: 1rem;
+    height: 1rem;
+    font-size: 1.1rem;
+    margin: 0 0 0.3rem 0.25rem;
+    :hover {
+      filter: opacity(25%);
+      cursor: pointer;
+    }
+  }
+`;
+const Time = styled.div`
+  font-size: 0.5rem;
+  margin-top: auto;
+  padding: 0.2rem;
+  margin-left: auto;
+`;
+const PartnerImage = styled.img`
+  height: 2.5rem;
+  width: 2.5rem;
+  margin-right: 0.2rem;
+  border-radius: 0.8rem;
+`;
+const PartnerName = styled.div`
+  font-size: 0.7rem;
+  padding: 0.2rem 0 0.3rem 0.5rem;
+`;
+const Content = styled.div`
+  display: flex;
+
+  .list.show {
+    margin-left: 0.5rem;
+    margin-top: auto;
+  }
+  .list.hide {
+    display: none;
+  }
+`;
+const Contents = styled.div`
+  display: block;
+
```

Comment:
```suggestion

  .isHover {
      display: none;
    }
  
  :hover{
      .isHover {
          display: block;
      }
  }
```

요렇게 hover 시에만 isHover 클래스가 나타나게 하면 CSS로도 가능해요!

Reply:
오!! 이런 방법이...!! 저도 저 부분의 코드를 작성하면서 이렇게 하는 것보다 더 간단하게 할 수 있을 것 같은데.... 일단 하자..! 라고 생각하고 다른 기능을 구현하느라 놓쳤던 부분인데!! 좋은 정보 알려주셔서 감사합니다ㅎㅎ :)